From 65a8a422a177548dd24831ff509ef05b29442c8d Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Tue, 9 Jun 2020 10:31:55 +0200
Subject: [PATCH] Enable watchdog reset

MPSoC watchdog requires PMU support to actually perform a reset. Set the ENABLE_EM
and ENABLE_RECOVERY_RESET_SYSTEM options to enable watchdog.

see:
https://xilinx-wiki.atlassian.net/wiki/spaces/A/pages/18842055/Cadence+WDT+Driver

Set the default action to POR since the default soft reset doesn't actually work.
---
 lib/sw_apps/zynqmp_pmufw/src/xpfw_config.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/sw_apps/zynqmp_pmufw/src/xpfw_config.h b/lib/sw_apps/zynqmp_pmufw/src/xpfw_config.h
index 5b8a860b98..38767742ab 100644
--- a/lib/sw_apps/zynqmp_pmufw/src/xpfw_config.h
+++ b/lib/sw_apps/zynqmp_pmufw/src/xpfw_config.h
@@ -148,7 +148,7 @@ extern "C" {
 #endif
 
 #ifndef ENABLE_EM_VAL
-#define	ENABLE_EM_VAL						(0U)
+#define	ENABLE_EM_VAL						(1U)
 #endif
 
 #ifndef ENABLE_SCHEDULER_VAL
@@ -160,11 +160,11 @@ extern "C" {
 #endif
 
 #ifndef ENABLE_RECOVERY_VAL
-#define	ENABLE_RECOVERY_VAL					(0U)
+#define	ENABLE_RECOVERY_VAL					(1U)
 #endif
 
 #ifndef ENABLE_RECOVERY_RESET_SYSTEM_VAL
-#define	ENABLE_RECOVERY_RESET_SYSTEM_VAL	(0U)
+#define	ENABLE_RECOVERY_RESET_SYSTEM_VAL	(1U)
 #endif
 
 #ifndef ENABLE_RECOVERY_RESET_PS_ONLY_VAL
-- 
2.43.0

